controllers/apidefinition_controller.go

```diff
@@ -113,6 +113,22 @@ func (r *ApiDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 		util.AddFinalizer(desired, keys.ApiDefFinalizerName)
 
+		if len(desired.Spec.UpstreamCertificateRefs) != 0 {
+			for domain, certName := range desired.Spec.UpstreamCertificateRefs {
+				tykCertID, err := r.checkSecretAndUpload(ctx, certName, namespacedName, log, &env)
+				if err != nil {
+					// we should log the missing secret but we should still create the API definition
+					log.Info(fmt.Sprintf("cert name %s is missing", certName))
+				} else {
+					if upstreamRequestStruct.Spec.UpstreamCertificates == nil {
```

Comment:
Maybe we can move this `if` check out of the `for` loop. Thus, instead of checking this condition on each condition that comes in the else statement, we can check it once on the outside of the for loop. Because once we have `UpstreamCertificates` map, we do not need further checks to verify it is not `nil`.

Reply:
Assuming no secrets are yet applied, this would just create an empty map regardless if there are secrets applied or not. In my opinion, we should only create a map when we have elements to put into it.